parking/src/parking.py

```diff
@@ -0,0 +1,172 @@
+
+class ParkingStation:
+    '''Class for Parking station to add or remove cars,
+	   also for extending the Levels, updating tariff plans'''
+    def __init__(self):
+        '''Constructor method'''
+        self.jsonDec = json.decoder.JSONDecoder()
+	
+    def __displayLevels(self):
+        ''' Returns the existing parking levels in the parking station.'''
+        levelInfoDict = {"LevelInfo" : []}
+
+        # Retrieve the parking level details from each and every level.
+        for level in ParkingLevel.objects.all().values("level_num","free_spots","occupied_spots","total_spots"):
+            levelInfoDict["LevelInfo"].append(level)
+        return levelInfoDict
+		
+    def deleteLevel(self,level_name):
+        ''' Delete or close a specific parking level from the station.'''
+        ParkingLevel.objects.filter(level_num=level_name).delete()
+        
+    def addLevel(self,level_name,total_spots):
+        '''Add or extend a new parking llevel to the station.'''
+        if len(ParkingLevel.objects.filter(level_num=level_name)) > 0:
+            # Check whether a parking level with the same name already exist.
+            return parkingExceptionsDict["LevelWithTheSameName"]
+
+        # Update the free spots and occupied spots based on total spots.
+        free_spots  = list(range(1,int(total_spots)+1))
+        occupied_spots = []
+        
+        # Create a new level with the details provided.
+        level = ParkingLevel(level_num=level_name,total_spots=total_spots,free_spots=json.dumps(list(free_spots)),occupied_spots=json.dumps(list(occupied_spots)))
+        level.save()
+        return self.__displayLevels()
+
+    def __getAvailableLocation(self):
+        '''Retrieves and returns an available spot by searching from all the available Levels.'''
+        location = None
+        if len(ParkingLevel.objects.all()) == 0:
+            # Check whether atleast one parking level exist.
+            raise ParkingLevelDoesntExist
+        for level in ParkingLevel.objects.all():
+            # Searching for free spots in each parking level.
+            free_spots = list(self.jsonDec.decode(level.free_spots))
+            if len(free_spots) > 0:
+                # Assign a parking spot in this specific level if free spots available.
+                assignedSpot = free_spots[-1]
+                location = level.level_num + "_" + str(assignedSpot)
+                break
+        # Raise an exception if no space left in any of the parking level.
+        if location is None: raise NoSpaceLeftInParking
+        return location
+        
+    def __checkCarNo(self,car_num):
+        '''Check whether car with the name exist in the parking slot.'''
+        try:
+            # Retrieving the car details with the name.
+            car = carDataDetails.objects.get(carno=car_num)
+        except carDataDetails.DoesNotExist:
```

Comment:
between not catching `carDataDetails.DoesNotExis` and `except carDataDetails.DoesNotExist: pass`, the second one is the worst option, please take care of this anti-pattern.